match/lib/match/storage/git_storage.rb

```diff
@@ -223,6 +185,26 @@ def add_user_config(user_name, user_email)
           end
         end
       end
+
+      private # rubocop:disable Lint/UselessAccessModifier
+
+      def git_push(commands: [], commit_message: nil)
+        commit_message ||= generate_commit_message
```

Comment:
Instead of doing an `||=` here, could you make `generate_commit_message` be the default value for `commit_message` please?

```ruby
def git_push(commands: [], commit_message: generate_commit_message)
  # ...
end
```

Reply:
That wouldn't work - `commit_message` already has a `nil` default in the methods up the call stack, it would end up being passed with `nil` here, meaning `generate_commit_message` wouldn't be called in your example. In pry:

```
[1] pry(main)> def a(test: nil)
[1] pry(main)*   b(test: test)
[1] pry(main)* end
=> :a
[2] pry(main)> def b(test: c)
[2] pry(main)*   puts test
[2] pry(main)* end
=> :b
[3] pry(main)> def c
[3] pry(main)*   "Hello"
[3] pry(main)* end
=> :c
[4] pry(main)> a

=> nil
```

Reply:
Oh, oops my bad. I didn't think about that. 👍